lib/string_decoder.js

```diff
@@ -72,9 +72,11 @@ StringDecoder.prototype.write = function write(buf) {
   if (typeof buf === 'string')
     return buf;
   if (!ArrayBuffer.isView(buf))
-    throw new ERR_INVALID_ARG_TYPE('buf',
-                                   ['Buffer', 'Uint8Array', 'ArrayBufferView'],
```

Comment:
Why is `Uint8Array` being removed here?

Reply:
[From the suggestion of this comment in #1826](https://github.com/nodejs/node/issues/1826#issuecomment-410487110), "string_decoder" seems to be on the list `TypedArray`/`DataView` support only. After viewing the [fs PR](https://github.com/nodejs/node/pull/22150) and the [child_process](https://github.com/nodejs/node/pull/22409), I just gave my best attempt to follow the "_idea_" to handle string_decode.

Please ;et me know if I misunderstood too much off from the context of the checklint & goals from the comment.

Reply:
I think this is okay and somewhat consistent with how we handle it elsewhere

Reply:
Hi @BeniCheni , I think broadly we can categorize the changes for these PRs into:
1. adding missing tests, for `TypedArray`s and `DataView`s wherever only `Buffer`s or `Uint8Array`s were tested earlier
2. Updating code to work with these new types, where necessary(should mostly involve only JS side of things, in `lib` and maybe `lib/internal`)
3. Updating documentation to reflect these changes.

Here, your PR is updating the errors being thrown, but is neither adding test cases nor adding functionality to work with the `TypedArray`s or `DataView`s. Please note, it might well be the case that `string_decoder` already works for these, but adding test cases is a way of validation for the same.  Hope that helps!

Reply:
🙏 for the collective reviews. Will analyze & update accordingly.